src/hooks/useSyncChainQuery.ts

```diff
@@ -31,8 +41,23 @@ export default function useSyncChainQuery() {
   const [searchParams, setSearchParams] = useSearchParams()
 
   useEffect(() => {
-    if (isActive && urlChainId && chainId !== urlChainId) {
+    // Change a user's chain on pageload if the connected chainId does not match the query param chain
+    if (isActive && urlChainId && chainIdRef.current === chainId && chainId !== urlChainId) {
       selectChain(urlChainId)
     }
-  }, [urlChainId, selectChain, searchParams, setSearchParams, isActive, chainId])
+    // If a user has a connected wallet and has manually changed their chain, update the query parameter if it's supported
+    else if (account && chainIdRef.current !== chainId && chainId !== urlChainId) {
+      if (isSupportedChain(chainId)) {
+        searchParams.set('chain', CHAIN_IDS_TO_NAMES[chainId])
+      } else {
+        searchParams.delete('chain')
+      }
+      setSearchParams(searchParams)
+    }
+
+    // If a user has a connected wallet and the chainId matches the query param chain, update the chainIdRef
+    if (isActive && chainId === urlChainId) {
+      chainIdRef.current = urlChainId
+    }
```

Comment:
Can this be moved into the `else` clause above? Other than that, lgtm.

Reply:
It can't because the `else if` around the `else` clause checks for the web3react chainId being different than the chainId from the url query parameter (`chainId !== urlChainId`) whereas this condition checks if they're equal.

I can make this an `else if` as well however which I've done in the latest commit.